include/etcpal/cpp/log.h

```diff
@@ -112,6 +115,67 @@ namespace etcpal
 /// }
 /// \endcode
 
+/// \ingroup etcpal_cpp_log
+/// \brief An object representing the current local time with millisecond resolution for logging purposes.
+/// \details Can also be invalid, which means it does not hold a valid timestamp.
+class LogTimestamp
+{
+public:
+  /// Construct an invalid timestamp by default.
+  LogTimestamp() = default;
+  constexpr LogTimestamp(unsigned int year, unsigned int month, unsigned int day, unsigned int hour,
+                         unsigned int minute, unsigned int second, unsigned int msec = 0, int utc_offset = 0);
+
+  bool IsValid() const noexcept;
+
+  constexpr const EtcPalLogTimestamp& get() const noexcept;
+  ETCPAL_CONSTEXPR_14 EtcPalLogTimestamp& get() noexcept;
+
+  static LogTimestamp Invalid();
+
+private:
+  EtcPalLogTimestamp timestamp_{};
```

Comment:
This is trivial, but I would think the trailing {} aren't required.

Reply:
Really? That's the initializer... if it's removed, wouldn't the POD struct member be uninitialized by the default constructor?

Reply:
I've played with godbolt -- you're absolutely right!  I hadn't considered initializers with POD -- very cool!